src/Ratios.jl

```diff
@@ -57,4 +59,15 @@ end
 
 decompose(x::SimpleRatio) = x.num, 0, x.den
 
+function __init__()
+    @require FixedPointNumbers = "53c48c17-4a7d-5ca2-90c5-79b7896eea93" begin
+        using .FixedPointNumbers: FixedPoint, Fixed, Normed, rawone
+        rawone_noerr(::Type{Fixed{T,f}}) where {T,f} = 1 << f
+        rawone_noerr(::Type{N}) where N<:Normed = rawone(N)
+        Base.promote_rule(::Type{SimpleRatio{S}}, ::Type{<:FixedPoint{T}}) where {S<:Integer,T<:Integer} = SimpleRatio{promote_type(S, T)}
+        SimpleRatio{S}(x::FixedPoint) where S<:Integer = SimpleRatio{S}(reinterpret(x), rawone_noerr(typeof(x)))
```

Comment:
How about inferring the parameter `S` from the `Fixed` type?
```
SimpleRatio(x::Normed{S}) where S = SimpleRatio{S}( x )
SimpleRatio(x::Fixed{S}) where {S <: Signed} = SimpleRatio{T}( -reinterpret(x), typemin(S) )
```

Reply:
We could add that. For the particular use here the promotion specifies the specific type and so of course you need to respect that, but for anyone who just calls `SimpleRatio(x::Fixed)` this would make sense.